### DISCUSSION TITLE
UNKNOWN: unknown error to open .next\static\chunks\app\layout.js on Windows

### DISCUSSION THREAD

**Discussion by jayjang0209** [vercel/next.js#60185] [Help]
### Summary

I encountered the error below in some events such as clicking <Link> Tag or reloading changes in the code. 
I tested on two different Windows machines and one Macbook, On both Windows machines, I had the same error. 
(and tested with both Nestjs13 and 14, seem the version doesn't matter)
It works fine on first running `npm run dev`, however, few minutes later, it keep raising this error and local server gets down

What I've tried so far as below, but none of them works
- removed .next folder and retry npm run dev
- opened visual studio code as admin
- ran dev with pnpm
- changed file permissions for project folder

```
PS C:\dev\test-nextjs\nextjs-init\portfolio-rev> npm run dev

> portfolio-rev@0.1.0 dev
> next dev

   ▲ Next.js 14.0.4
   - Local:        http://localhost:3000

 ✓ Ready in 4.6s
 ○ Compiling / ...
 ✓ Compiled / in 3.5s (1124 modules)
 ⨯ [Error: UNKNOWN: unknown error, open 'C:\dev\test-nextjs\nextjs-init\portfolio-rev\.next\static\chunks\app\layout.js'] {
  errno: -4094,
  code: 'UNKNOWN',
  syscall: 'open',
  path: 'C:\\dev\\test-nextjs\\nextjs-init\\portfolio-rev\\.next\\static\\chunks\\app\\layout.js'
}
[Error: UNKNOWN: unknown error, open 'C:\dev\test-nextjs\nextjs-init\portfolio-rev\.next\static\chunks\app\layout.js'] {
  errno: -4094,
  code: 'UNKNOWN',
  syscall: 'open',
  path: 'C:\\dev\\test-nextjs\\nextjs-init\\portfolio-rev\\.next\\static\\chunks\\app\\layout.js'
}
[Error: UNKNOWN: unknown error, open 'C:\dev\test-nextjs\nextjs-init\portfolio-rev\.next\static\chunks\app\layout.js'] {
  errno: -4094,
  code: 'UNKNOWN',
  syscall: 'open',
  path: 'C:\\dev\\test-nextjs\\nextjs-init\\portfolio-rev\\.next\\static\\chunks\\app\\layout.js'
}
```
![image](https://github.com/vercel/next.js/assets/77429796/d9d9e517-4e15-4d7e-8c60-d759b67df3c3)

### Additional information
package.json

```js
{
  "name": "portfolio-rev",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "@types/node": "20.10.6",
    "@types/react": "18.2.46",
    "@types/react-dom": "18.2.18",
    "autoprefixer": "10.4.16",
    "eslint": "8.56.0",
    "eslint-config-next": "14.0.4",
    "framer-motion": "^10.17.0",
    "next": "14.0.4",
    "postcss": "8.4.32",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-icons": "^4.12.0",
    "tailwindcss": "3.4.0",
    "typescript": "5.3.3"
  }
}
```

**Comment by genexyz** [comment]
This is constantly happening to me too.

Any fix yet?

**Comment by JulieChabanis** [comment]
Same for me...

**Comment by DevDaniloFerrari** [comment]
What version of node are you using?

**Comment by stanislavyv** [comment]
Same for me

**Comment by ObaidAshiq** [comment]
have u guys tried canary version ?
Which version are u guys using
@jayjang0209  @genexyz  @DevDaniloFerrari  @stanislavyv  @JulieChabanis

**Reply by stanislavyv** [reply]
I've tried on 14.0.4 and 13.5.5. Node v18.18.0. For me it started happening recently, I was using Next 13.5.5 for months without such error occurring.

**Comment by DevDaniloFerrari** [comment]
This error generally happens when you import a file but the casing is wrong.

**Reply by GUSILLUS** [reply]
Can you explain in more detail, please?

**Reply by ObaidAshiq** [reply]
What he meant is, the imports that u have made, might  not be correct in terms of spelling.
Check them

**Reply by DevDaniloFerrari** [reply]
Exactly

**Reply by GUSILLUS** [reply]
![image](https://github.com/vercel/next.js/assets/97075591/ee2239a4-4387-49cd-a38b-59fda90da754)
Everything is ok, can you help to fix it?

**Reply by ObaidAshiq** [reply]
Quick fix : is stop your development server
Close the running instance of your project on the browser
Delete the .next folder
Restart your ide (VS code)
pray that its fixed

**Reply by GUSILLUS** [reply]
Yep, but i need to do it always)

**Reply by PunitSoniME** [reply]
Automate removing .next folder when project starts

`"dev": "rimraf .next && next dev",`

**Comment by harmadema** [comment]
Since a few weeks I get this error to when running dev. Whenever I make a change in the project the fast refresh gets stuck loading, if I then do a refresh in the browser it crashes and the error appears. My colleges dont have this issue and are all working in the samen git repo. So it looks like its something on my pc...

**Reply by ObaidAshiq** [reply]
i would suggest delete that .next folder thats get generated on build time, and do a hard reset of ur Developer ide and close your browser that is having your development window

**Reply by harmadema** [reply]
Thanks for the suggestion. When trying to reset vscode I saw there was a update ready. After updating the refreshing worked again (for now). Maybe resetting vscode by deleting the settings json and restarting the editor works for anyone else.

**Reply by harmadema** [reply]
It was a short solution in 2 days it didn't work anymore

**Comment by sFaxsy** [comment]
Same for me since the last update

**Comment by alohe** [comment]
I commented out the changes i made a while back in the `next.config.js` file until it started working. 

Here is what i did:

👇 Before 
![image](https://github.com/vercel/next.js/assets/50866431/9088c97f-a810-4e65-a385-e36d0f2492cd)

👇 After
![image](https://github.com/vercel/next.js/assets/50866431/acaa3c84-9483-4b52-996f-76a731dedde2)

And it works now 🤷‍♂️, also did what the others on here suggested, deleting the .next folder and restarting vs-code which didn't work.
It works initially but when i make a change to a file while re-compiling it throws `Internal server error`.

So that's when i decided to start commenting out the changes i made to the `next.config.js` file in **next 13** and update the the **images.domains** code to **next 14** to **images.remotePatterns**. while also removing `swcMinify: true` which for some reason made it work. 

Hope this helps, I'll be sure to report back if this stops working at some point.

**Comment by erkamdemirci** [comment]
still getting this error constantly, v14.2.1

**Comment by stanislavyv** [comment]
In my case, it was the antivirus that was constantly killing my dev server (win 11 + Bitdefender). After adding my project directory to the [exceptions list](https://www.bitdefender.com/consumer/support/answer/13427/), the error disappeared, and performance also vastly improved.

**Reply by edocollado** [reply]
same for my 💯

**Reply by Joetech12** [reply]
Worked, thanks.

**Reply by abhishekkumbhani** [reply]
Thank you!
Disabling an antivirus does the job for me. :)

**Reply by harmadema** [reply]
In my case bitdefender is managed from a outside company. They added the exclusion and something more and now it works finally!!!

**Comment by dydcodydco** [comment]
This is constantly happening to me too.

Any fix yet?

**Reply by Trabontea** [reply]
The solution is to add your next.js folder to the exclusion list of antivirus or other software for malware, in my case  Emsisoft.

**Reply by Choiuijin1125** [reply]
I also solved this problem after I stopped Riot Vanguard... :)
